### ISSUE TITLE
Module is not able to handle pipline

### ISSUE THREAD

**Issue by jkiani64** [oegedijk/explainerdashboard#100]
Imagine that you want to use a pipeline that performs feature engineering and data cleaning and then training your model. This could be useful for the production level as the new data comes it goes through the pipeline for cleaning, building features  (based on the training data), and predicting its label. This module is not able to handle such a case. Adding such an ability could be very useful.

Here is a very simple example that the module is not able to handle and throws an error. I tried to log transform the target variable and then fit the model. I can send more examples upon request.

````
mdl = TransformedTargetRegressor(regressor=RandomForestRegressor(),
                                 func=np.log1p, inverse_func=np.exp,
                                check_inverse=False)

explainer = RegressionExplainer(mdl, X_test, y_test, 
                                shap = 'kernel',
                                cats=['Sex', 'Deck', 'Embarked'], 
                                idxs=test_names, 
                                descriptions = feature_descriptions,
                                target='Fare',
                                units="$")

ExplainerDashboard(explainer).run()
````

**Comment by oegedijk** [comment]
Hi @jkiani64, the general problem with sklearn pipelines is that they do not keep track of feature names: inside the pipeline numpy arrays get passed around which means that by the time you reach the final model you have no idea which columns belong to what features, which makes it hard to build a feature explainer!

However pipelines that do not change the number or order of columns are supported: I just run the pipeline until the final step, take that as the input df and assign the column names from the input df, and then extract the model.

It seems like TransformedTargetRegressor could be supported in a similar way. Do you have other examples?

**Comment by mikaelmv** [comment]
Hi @oegedijk , 

I am experiencing some conceptual issues with pipelines and transformations in general.

I am applying a MinMaxScaler() transformation to my numerical features and the explainer correctly shows up with the expected predictions and performance measures (which is great). 

However the features are now set to the transformed scale and therefore it is not very practical/interpretable to play with the new units in the "What if?" section. On the other side if I don't transform the variables, then predictions are effected in the previous outputs as model underperforms with the original units. Seems like a chicken or egg problem. 

Is there a suggested solution? Thank you and thanks again !

**Comment by oegedijk** [comment]
ah, if you are using xgboost then there should be no need to scale your features! 

Scaling is only really useful for inputs into neural networks or with linear regression when you want to interpret the size of coefficients. Tree based methods don't need feature scaling. 

Are you sure the scaling increases performance?

**Comment by oegedijk** [comment]
Should work better now, try the latest release: https://github.com/oegedijk/explainerdashboard/releases/tag/v0.3.3
